cg/services/order_validation_service/workflows/microsalt/validation/inter_field/utils.py

```diff
@@ -0,0 +1,35 @@
+from cg.models.orders.sample_base import ContainerEnum
+from cg.services.order_validation_service.errors.sample_errors import OccupiedWellError
+from cg.services.order_validation_service.workflows.microsalt.models.order import MicrosaltOrder
+from cg.services.order_validation_service.workflows.microsalt.models.sample import MicrosaltSample
+
+
+class PlateSamples:
+
+    def __init__(self, order: MicrosaltOrder):
+        self.wells: dict = {}
+        self._initialize_map(order)
+
+    def _initialize_map(self, order: MicrosaltOrder):
+        for sample_index, sample in order.enumerated_new_samples:
+            if is_sample_on_plate(sample):
+                key: tuple[str, str] = (sample.container_name, sample.well_position)
+                if not self.wells.get(key):
+                    self.wells[key] = []
+                self.wells[key].append(sample_index)
+
+    def get_occupied_well_errors(self) -> list[OccupiedWellError]:
```

Comment:
For me it is not super clear that this method should be in this class, at least not from the naming. But feel free to keep it. Just thought I'd mention it.

Reply:
The class name sounds like a data class to me

Reply:
I renamed it to PlateSamplesValidator. I think it is a bit better.